map_view/display-device-location-with-autopan-modes/src/main/java/com/esri/samples/display_device_location_with_autopan_modes/DisplayDeviceLocationWithAutopanModesSample.java

```diff
@@ -143,17 +159,13 @@ public void start(Stage stage) {
         }
       });
 
-      // enable the checkbox and combo box interactions when the map is loaded
-      map.addDoneLoadingListener(() -> {
-        if (map.getLoadStatus() == LoadStatus.LOADED) {
-          checkBox.setDisable(false);
-          checkBox.setSelected(true);
-          comboBox.setDisable(false);
-
-          // start the location display
-          locationDisplay.startAsync();
-        } else {
-          new Alert(Alert.AlertType.ERROR, "Map failed to load: " + map.getLoadError().getCause().getMessage()).show();
+      mapView.setOnMousePressed(event -> {
```

Comment:
nearly! With mouse pressed, a user can press and then click away, and the drop down box won't be updated. 

This is because when the user presses, the locationDisplay mode is still active. When they release (click), the locationDisplay is off but the comboBox still shows the results of the first press. The second press picks up the locationDisplay off, and then the release (click) updates the combobox). 

We can simply change this to `setOnMouseClicked` so that when the user clicks, it updates the combobox correctly. 

```suggestion
      mapView.setOnMouseClicked(event -> {
```